### ISSUE TITLE
Rushstack Doc Revamp Roadmap

### ISSUE THREAD

**Issue by elliot-nelson** [microsoft/rushstack.io-website#36]
### Intro

As rush (and rushstack) grow in popularity, we want to make sure docs are accessible, accurate, and easy to contribute to. In particular, it would be nice if it was easier to link back and forth between Heft, Rush, and API Extractor, and if the overall rush ecosystem was more discoverable from our docs pages.

To accomplish this, the plan is to:

1. Convert all existing doc sites, individually, to a TypeScript-based documentation framework.  Our current plan is to use Docusaurus for this revamp.
2. Combine all doc sites into a new Rush monorepo, adjusting deployment workflows as necessary.  (Each site will retain its current layout and URL, they will just all have their content source in one repo.)
3. Design a new overall layout for the header and navigation that allows users to easily understand how each site fits into the rushstack ecosystem.

#### Part 1 (Convert to Docusaurus)

The plan is to individually convert each doc site to Docusaurus.  We'll start with `rushstack.io`, because it contains a large API Reference section and will likely require the most effort (if there are any issues or caveats to our use of Docusaurus, we want to flush them out early).  Once `rushstack.io` is complete we'll move on to the rushjs and api-extractor sites.

#### Part 2 (Monorepo)

TBD: What will the repo name be for the new monorepo? `microsoft/rushstack-docs`?

TBD: Hosting.  (Do we have `microsoft/rushstack-docs` output dist output that is copied into the existing repos, and let GitHub Pages do its thing like today? Do we investigate a more complete hosting solution at another provider?)

#### Part 3 (Design)

When we reach this point, we'll compile a list of our needs for the new combined header design and work from there.

### Future Work

We hope to get some additional knock-on effects from this conversion:

1. A single repo users can file issues about missing or requested documentation (just like the single rushstack repo for implementation).
2. Possibility of translating / internationalization, using Docusaurus's existing support.
3. Possibility of providing _versioned_ docs/API references, if appropriate, using Docusaurus's existing support.
4. ???

**Comment by elliot-nelson** [comment]
*Link Index*

 - https://github.com/microsoft/rushstack.io-website/pull/37 - MVP Docusaurus changes for rushstack.io
 - https://github.com/microsoft/rushstack/pull/2979 - Docusaurus plugin for API Documenter

**Comment by octogonz** [comment]
Closing, since these goals are now accomplished.
